tensorflow_datasets/scripts/benchmark_dataset.py

```diff
@@ -0,0 +1,100 @@
+r"""Benchmark dataset.
+
+python -m tensorflow_datasets.scripts.benchmark_dataset \
+  --dataset dataset_name
+
+"""
+import time
+from typing import Dict
+
+from absl import app
+from absl import flags
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets as tfds
+
+# DEFAULT_DATA_DIR = os.path.expanduser(os.path.join("~", "tensorflow_datasets"))
+
+EPOCHS = 1
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('dataset', None, 'Dataset name')
+flags.DEFINE_string(
+    'data_dir', None,
+    'Path to dataset directory (auto-computed). Dataset should already present')
+flags.DEFINE_integer('epochs', EPOCHS, 'No. of iterations over the dataset')
+
+
+def _preprocess(
+    name: str,
+    data_dir: str,
+) -> Dict[str, tf.data.Dataset]:
+
+  # Exclude dataset download time
+  builder = tfds.builder(name, data_dir=data_dir)
+  builder.download_and_prepare()
+
+  start_time = time.perf_counter()
+  ds = builder.as_dataset()
+  print('\nTime taken to preprocess {} dataset: {}\n'
+        .format(name, time.perf_counter() - start_time))
+
+  return ds
+
+
+def benchmark(
+    dataset: tf.data.Dataset,
+    epochs: int,
+) -> None:
+  """Benchmarks the dataset.
+
+  Reports:
+  - Total Execution time
+  - Time taken per Epoch
+  - Dataset elements per second
+  """
+
+  num_examples = tf.data.experimental.cardinality(dataset)
+  if num_examples == tf.data.experimental.UNKNOWN_CARDINALITY:
+    num_examples = sum(1 for _ in dataset)
```

Comment:
Let's not iterate twice over the full dataset as it may takes multiple hours so we don't want to double the time.